streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/HandlingSourceTopicDeletionIntegrationTest.java

```diff
@@ -124,12 +127,15 @@ public void shouldThrowErrorAfterSourceTopicDeleted(final boolean useNewProtocol
 
         CLUSTER.deleteTopic(INPUT_TOPIC);
 
-        TestUtils.waitForCondition(
-            () -> kafkaStreams1.state() == State.ERROR && kafkaStreams2.state() == State.ERROR,
-            TIMEOUT,
-            () -> "Kafka Streams clients did not reach state ERROR"
-        );
+        if (!useNewProtocol) {
+            TestUtils.waitForCondition(
+                    () -> kafkaStreams1.state() == State.ERROR && kafkaStreams2.state() == State.ERROR,
+                    TIMEOUT,
+                    () -> "Kafka Streams clients did not reach state ERROR"
+            );
 
+        }
+        TimeUnit.SECONDS.sleep(5);
         assertThat(calledUncaughtExceptionHandler1.get(), is(true));
         assertThat(calledUncaughtExceptionHandler2.get(), is(true));
```

Comment:
How about doing this instead of always sleeping for a fixed time

```suggestion
TestUtils.waitForCondition(
        () -> calledUncaughtExceptionHandler1.get() && calledUncaughtExceptionHandler2.get(),
        TIMEOUT,
        () -> "......Error Message......"
);
```